src/core/database.py

```diff
@@ -29,29 +29,35 @@ def dbName(self):
         return DB_NAME
 
     @property
-    def _layerName(self, directory):
+    def _layerName(self):
         return LAYER_NAME
 
     def createEmptyDataBase(self, directory):
-        geom = QgsWkbTypes.Point
-        tableName = "emptyTable"
         db = self._pathToDataBase(directory)
 
-        fields = QgsFields()
+        self.AddTableToDataBaseFile(
+            db, self._createFieldsForEnvironmentTable(), "environmentTable"
+        )
 
+    def AddTableToDataBaseFile(self, db, fields, tableName):
+        """
+        Create the database and save it as gpkg file
+
+        :param db: path to gpkg file to append to or create
+        :param fields: the fields of the table
+        :param tableName: the name of the table
+        """
+        geom = QgsWkbTypes.Point
```

Comment:
The geometry depends on the underlying data. In case of the environment table, there aren't any geometries so its `QgsWkbTypes.NoGeometry`. So the geometry is also a parameter at some point.